File: DrissionPage/_units/cookies_setter.py
# -*- coding:utf-8 -*-
from http.cookiejar import Cookie

from .._functions.web import set_browser_cookies, set_session_cookies


class CookiesSetter(object):
    def __init__(self, page):
        self._page = page

    def __call__(self, cookies):
        """设置一个或多个cookie
        :param cookies: cookies信息
        :return: None
        """
        if (isinstance(cookies, dict) and 'name' in cookies and 'value' in cookies) or isinstance(cookies, Cookie):
            cookies = [cookies]
        set_browser_cookies(self._page, cookies)

    def remove(self, name, url=None, domain=None, path=None):
        """删除一个cookie
        :param name: cookie的name字段
        :param url: cookie的url字段，可选
        :param domain: cookie的domain字段，可选
        :param path: cookie的path字段，可选
        :return: None
        """
        d = {'name': name}
        if url is not None:
            d['url'] = url
        if domain is not None:
            d['domain'] = domain
        if path is not None:
            d['path'] = path
        self._page.run_cdp('Network.deleteCookies', **d)

    def clear(self):
        """清除cookies"""
        self._page.run_cdp('Network.clearBrowserCookies')


class SessionCookiesSetter(object):
    def __init__(self, page):
        self._page = page

    def __call__(self, cookies):
        """设置多个cookie，注意不要传入单个
        :param cookies: cookies信息
        :return: None
        """
        if (isinstance(cookies, dict) and 'name' in cookies and 'value' in cookies) or isinstance(cookies, Cookie):
            cookies = [cookies]
        set_session_cookies(self._page.session, cookies)

    def remove(self, name):
        """删除一个cookie
        :param name: cookie的name字段
        :return: None
        """
        self._page.session.cookies.set(name, None)

    def clear(self):
        """清除cookies"""
        self._page.session.cookies.clear()

File: java/src/main/java/com/ll/DrissonPage/units/cookiesSetter/CookiesSetter.java
package com.ll.DrissonPage.units.cookiesSetter;

import com.ll.DrissonPage.functions.Web;
import com.ll.DrissonPage.page.ChromiumBase;
import lombok.AllArgsConstructor;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * @author 陆
 * @address <a href="https://t.me/blanksig"/>click
 */
@AllArgsConstructor
public class CookiesSetter {
    private final ChromiumBase page;

    /**
     * 删除一个cookie
     *
     * @param name cookie的name字段
     */
    public void remove(String name) {
        remove(name, null);
    }

    /**
     * 删除一个cookie
     *
     * @param name cookie的name字段
     * @param url  cookie的url字段，可选
     */
    public void remove(String name, String url) {
        remove(name, url, null);
    }

    /**
     * 删除一个cookie
     *
     * @param name   cookie的name字段
     * @param url    cookie的url字段，可选
     * @param domain cookie的domain字段，可选
     */
    public void remove(String name, String url, String domain) {
        remove(name, url, domain, null);
    }

    /**
     * 删除一个cookie
     *
     * @param name   cookie的name字段
     * @param url    cookie的url字段，可选
     * @param domain cookie的domain字段，可选
     * @param path   cookie的path字段，可选
     */
    public void remove(String name, String url, String domain, String path) {
        Map<String, Object> map = new HashMap<>();
        map.put("name", name);
        if (url != null && !url.isEmpty()) map.put("url", url);
        if (domain != null && !domain.isEmpty()) map.put("domain", url);
        if (path != null && !path.isEmpty()) map.put("path", url);
        this.page.runCdp("Network.deleteCookies", map);
    }

    public void add(Map<String, Object> cookies) {
        List<Map<String, Object>> cookies1 = new ArrayList<>();
        cookies1.add(cookies);
        Web.setBrowserCookies(this.page, cookies1);
    }

    /**
     * 清除cookies
     */
    public void clear() {
        this.page.runCdp("Network.clearBrowserCookies");
    }
}
